segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -142,17 +138,32 @@ public void setupAppend(SetupAppend setupAppend) {
      * that is written.
      */
     public void performNextWrite() {
-        Append append;
+        Append append = getNextAppend();
+        if (append == null) {
+            return;
+        }
 
-        synchronized (lock) {
-            if (outstandingAppend != null || waitingAppends.isEmpty()) {
-                return;
+        Timer timer = new Timer();
+        storeAppend(append).whenComplete((v, e) -> {
+            handleAppendResult(append, e);
+            if (e == null) {
+                WRITE_STREAM_SEGMENT.reportSuccessEvent(timer.getElapsed());
+            } else {
+                WRITE_STREAM_SEGMENT.reportFailEvent(timer.getElapsed());
             }
+            append.getData().release();
```

Comment:
is this `release` a cleanup task? if so, are there any chances that the code just above it might fail and thus this would never get called?